src/main/java/com/faforever/client/remote/FafServerAccessorImpl.java

```diff
@@ -471,10 +471,42 @@ private void onServerPing() {
 
   private void dispatchAuthenticationFailed(AuthenticationFailedMessage message) {
     fafConnectionTask.cancel();
-    loginFuture.completeExceptionally(new LoginFailedException(message.getText()));
+
+    String text = translateAuthenticationMessage(message);
+
+    loginFuture.completeExceptionally(new LoginFailedException(text));
     loginFuture = null;
   }
 
+  private String translateAuthenticationMessage(AuthenticationFailedMessage message) {
```

Comment:
So this ends in a lot of lines if we do this for every mesage. What I would expect would be a unique identifying string/number for the message. Which in return can be found in the messages of the client maybe with a prefix For Example:
Server message has a field messageId= 147
-> i18n.get("serverMessage.147", message.getParameters());
-> serverMessage.147 = Login failed, you are banned for {0} days
Translating hence can be done in one line

Reply:
Ok I see. I'll have to think about the best way to do this so it still makes sense for the protocol. The only thing is it doesn't handle config parameters like the website url which don't need to be sent by the server.

Reply:
I don't think I would use numbers, but I might do something like `i18n.get(String.format("authenticationMessage.%s", message.getContext()), /* parameters here */)`

Reply:
Just saying the here presented solution seems to be a bad idea. Also what we definitely need is a fallback string that is sent if the server message is not found in message.properties

Reply:
Ok. I've made it more general where it uses the value of the message identifier as the localization key. So for instance localization of a policy server error for "fraudulent" response would use key `login.error.policy.fraudulent`.

If the given key is not found, it will try to fall back by popping off the last identifier in the chain until either it finds a key, or it hits `login.error` which returns a very generic error message. So if for instance the key `login.error.policy.fraudulent` didn't exist, it would try  `login.error.policy` which does exist, and returns a generic error about a policy violation.

Reply:
sounds better

Reply:
ok but still this is specific code for "login",  so we need to copy this for everything else right. Tho I understand that if u wanna add specific args from the clientProperties thats hard to realize. So if u want to do it this way that's fine with me.

Reply:
Well when I do the next message type I can refactor the bits I need.